### PR TITLE
[bitnami/kibana] add publicBaseUrl param

### PR DESCRIPTION
**Pull request by mike12806**
### Description of the change

This change maps a configuration item in server.publicBaseURL to the Kibana config file (server.publicBaseUrl). See https://www.elastic.co/guide/en/kibana/8.14/settings.html#server-publicBaseUrl

### Benefits

Kibana throws an annoying pop-up on each initial load asking you to set this variable. I'm sure it probably also provides a security benefit, but i'm primarily annoyed with the pop up :) 

### Possible drawbacks

There's no real drawbacks and arguably this could be added as a parameter in "extraConfiguration", however given Kibana is pushing this as an issue in the UI, it feels more appropriate to be an explicit configuration option IMO 

### Applicable issues

None

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kibana/templates/configmap.yaml [hunk 1]
```diff
@@ -22,6 +22,9 @@ data:
     {{- if .Values.configuration.server.basePath }}
     server.basePath: {{ .Values.configuration.server.basePath | quote }}
     {{- end }}
+    {{ if .Values.configuration.server.publicBaseUrl }}
+    server.publicBaseUrl: {{ .Values.configuration.server.publicBaseUrl | quote }}
+    {{- end }}
```
Comment (juan131) [inline bitnami/kibana/templates/configmap.yaml:27]: I wonder whether it makes sense to set this configuration property based on the ingress hostname (when provided). Sth like this:
```suggestion
    {{- if or .Values.configuration.server.publicBaseUrl .Values.ingress.enabled }}
    server.publicBaseUrl: {{ default .Values.ingress.hostname .Values.configuration.server.publicBaseUrl }}
    {{- end }}
```

**Review by juan131 — COMMENTED** [review]
Thanks so much for this PR @mike12806 !! Please take a look to my comments when you have a chance.

**Comment by mike12806** [comment]
> Thanks so much for this PR @mike12806 !! Please take a look to my comments when you have a chance.

Makes sense to me! Just accepted your change - thanks!

**Review by juan131 — APPROVED** [review]
LGTM

**Comment by ftqualifio** [comment]
> The publicly available URL that end-users access Kibana at. Must include the protocol, hostname, port (if different than the defaults for http and https, 80 and 443 respectively), and the [server.basePath](https://www.elastic.co/guide/en/kibana/current/settings.html#server-basePath) (if configured).

`.Values.ingress.hostname` doesn't include the protocol or port.

**Comment by RytisLT** [comment]
> > The publicly available URL that end-users access Kibana at. Must include the protocol, hostname, port (if different than the defaults for http and https, 80 and 443 respectively), and the [server.basePath](https://www.elastic.co/guide/en/kibana/current/settings.html#server-basePath) (if configured).
> 
> `.Values.ingress.hostname` doesn't include the protocol or port.

yep, this breaks my kibana deployment. I had `server.publicBaseUrl` as `extraConfiguration`, now it is set incorrectly because protocol is missing and kibana refuses to boot. Also I cannot change it with `extraConfiguration` because it creates duplicate key

**Pull request merged**